File: files/100-199/156/CH-156.R
library(tidyverse)
library(readxl)

path = "files/CH-156 Column Splitting.xlsx"
input = read_excel(path, range = "B2:B8")
test  = read_excel(path, range = "D2:F8")

result = input %>%
  mutate(ID.1 = ifelse(nchar(ID) %% 2 == 0, 
                       substr(ID, 1, nchar(ID)/2), 
                       substr(ID, 1, nchar(ID)/2)),
         ID.2 = ifelse(nchar(ID) %% 2 == 0, 
                       substr(ID, nchar(ID)/2 + 1, nchar(ID)), 
                       substr(ID, nchar(ID)/2 + 1, nchar(ID)/2 + 1)),
         ID.3 = ifelse(nchar(ID) %% 2 == 0, 
                       NA, 
                       substr(ID, nchar(ID)/2 + 2, nchar(ID)))) %>%
    select(-ID)

all.equal(result, test, check.attributes = FALSE)
# only one discrepancy from original solution                                                                                                               

File: files/100-199/156/CH-156.py
import pandas as pd

path = "CH-156 Column Splitting.xlsx"
input = pd.read_excel(path, usecols="B", skiprows=1, nrows=7)
test = pd.read_excel(path, usecols="D:F", skiprows=1, nrows=7).fillna('')

def split_id(id):
    n = len(id)
    mid = n // 2
    if n % 2 == 0:
        id1, id2, id3 = id[:mid], id[mid:], None
    else:
        id1, id2, id3 = id[:mid], id[mid:mid + 1], id[mid + 1:]
    return pd.Series([id1, id2, id3])

result = input['ID'].apply(split_id).fillna('')
result.columns = ['ID.1', 'ID.2', 'ID.3']

# identical except one field wrong in given solution